### PR TITLE
feat(email): add cyrillic support for email

### PR DESCRIPTION
**Pull request by zzorin**
https://jira.railsc.ru/browse/PC4-28327

### CODE REVIEW

File: spec/lib/apress/utils/email_validation_spec.rb [hunk 1]
```diff
@@ -19,10 +19,10 @@
   end
 
   context 'when email is valid' do
-    it_behaves_like('valid email', %w{valid.email@email.com va-l-id_email@email.com validemail@email.com valid.email@e-mail.com valid_email@e.mail.com v3alid_email01@e.mail.com })
+    it_behaves_like('valid email', %w{моя@почта.рф valid.email@email.com va-l-id_email@email.com validemail@email.com valid.email@e-mail.com valid_email@e.mail.com v3alid_email01@e.mail.com })
```
Comment (apress-hound) [inline spec/lib/apress/utils/email_validation_spec.rb:22]: Metrics/LineLength: Line is too long. [192/120]<br>Layout/SpaceInsidePercentLiteralDelimiters: Do not use spaces inside percent literal delimiters.

File: lib/apress/utils/email_validation.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 module Apress
   module Utils
     module EmailValidation
-      EMAIL_REGEXP = /\A[a-z0-9\-_]+(?:\.[a-z0-9\-\._]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z]{2,}\Z/i
+      EMAIL_REGEXP = /\A[a-zа-яё0-9\-_]+(?:\.[a-zа-яё0-9\-\._]+)*@(?:[a-zа-яё0-9](?:[a-zа-яё0-9-]*[a-zа-яё0-9])?\.)+[a-zа-яё0-9](?:[a-zа-яё0-9-]*[a-zа-яё0-9])?\Z/i.freeze
```
Comment (apress-hound) [inline lib/apress/utils/email_validation.rb:7]: Metrics/LineLength: Line is too long. [170/120]

File: spec/lib/apress/utils/email_validation_spec.rb [hunk 3]
```diff
@@ -19,10 +19,10 @@
...
   end
 
   context 'when email is invalid' do
-    it_behaves_like('invalid email', %w{inv&alidemail@email.com invalid.email@e_mail.com inv#alid-email@email.com envalid$email.com invalid@email invalid@email.96})
+    it_behaves_like('invalid email', %w{inv&alidemail@email.com invalid.email@e_mail.com inv#alid-email@email.com envalid$email.com invalid@email})
```
Comment (apress-hound) [inline spec/lib/apress/utils/email_validation_spec.rb:26]: Metrics/LineLength: Line is too long. [147/120]

File: spec/lib/apress/utils/email_validation_spec.rb [hunk 4]
```diff
@@ -19,10 +19,29 @@
...
-    it_behaves_like('invalid email', %w{inv&alidemail@email.com invalid.email@e_mail.com inv#alid-email@email.com envalid$email.com invalid@email invalid@email.96})
+    it_behaves_like(
+      'invalid email',
+      %w{
+        inv&alidemail@email.com
+        invalid.email@e_mail.com
+        inv#alid-email@email.com
+        envalid$email.com invalid@email
+        }
+      )
```
Comment (apress-hound) [inline spec/lib/apress/utils/email_validation_spec.rb:45]: Layout/ClosingParenthesisIndentation: Indent ) to column 4 (not 6)

File: spec/lib/apress/utils/email_validation_spec.rb [hunk 5]
```diff
@@ -19,10 +19,29 @@
...
+      %w{
+        моя@почта.рф
+        valid.email@email.com
+        va-l-id_email@email.com
+        validemail@email.com
+        valid.email@e-mail.com
+        valid_email@e.mail.com
+        v3alid_email01@e.mail.com
+        }
+      )
```
Comment (apress-hound) [inline spec/lib/apress/utils/email_validation_spec.rb:33]: Layout/ClosingParenthesisIndentation: Indent ) to column 4 (not 6)

**Review by taleksei — APPROVED** [review]
@isqad

**Pull request merged**